adagrams/game.py

```diff
@@ -1,11 +1,169 @@
+        returned_letters.append(current_letter)
+        letter_list.remove(current_letter)
+
+    return returned_letters
+    
 
 def uses_available_letters(word, letter_bank):
-    pass
+    """
+    Checks if the provided word utilizes letters only found in letter bank
+
+        Parameters:
+                word (str): the played word
+                letter_bank (list): bank of randomly assigned letters
+
+        Returns:
+                boolean value based on word validity
+    """
+    letter_dict = Counter(letter_bank)
+
+    for letter in word.upper():
+        try:
+            if letter in letter_dict.keys() and letter_dict[letter] > 0:
+                letter_dict[letter] -= 1
+            else:
+                return False
+        except KeyError:
+            return False
+    
+    return True
 
 def score_word(word):
-    pass
+    ''' 
+    Scores the word based on the points in the 
+    LETTER_POOL dictionary and the additional rule
+    if the word is longer than 6 characters then it
+    will add 8 to the total score.
+
+        Parameters: 
+                word(str): the word to be scored
+        Returns: 
+                The total score of the word. (int)
+    '''
+    score = 0
+    for letter in word:
+        score += LETTER_POOL[letter.upper()][1]
+    if len(word) > 6:
+        score += 8
+    return score
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+
+    '''
+    Calculates the highest score in a list of words.
+
+    Analyzes high scores to check for a tie. 
+    Follows game rules for tie breaker:
+
+    Ties are judged based off of letter count:
+
+    If Their is a 12 letter word present, returns the
+    first 12 letter word that appears in the list.
+
+    For cases without a 12 letter word, returns the 
+    shortest word in the list.
+
+    If the shortest words are the same length, return
+    the one that appears first in the original list.
+
+        Parameters:
+                List of words to be scored.
+        
+        Returns: 
+                A tuple of the highest scoring word followed
+                by its point value.
+
+    '''
+
+    scores = []
+    lowest = ()
+
+    for word in word_list:
+        current_word_score = score_word(word)
+        if not scores: 
+            scores.append((word, current_word_score))
+        elif current_word_score > scores[0][1]:
+            scores = [(word, current_word_score)]
+        elif current_word_score == scores[0][1]:
+            scores.append((word, current_word_score))
+
+    if len(scores) == 1:
+        return scores[0]
+
+    for word, top_score in scores:
+        word_length = len(word)
+        if word_length == 10:
+            return word, top_score
+        elif not lowest:
+            lowest = (word, top_score, word_length)
+        elif word_length < lowest[2]:
+            lowest = (word, top_score, word_length)
```

Comment:
You could also pull this into a helper method to simplify the logic in this function